wordpress-mobile/WordPress-iOS
# Adds Jetpack Add-ons to Zendesk tickets

Fixes #NA

[Related WordPressKit PR](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/414)

This PR updates the creation of a Zendesk ticket as follows:

1. The `plan`(and the priority of the ticket) sent to Zendesk will reflect the site in use.
2. When opening a ticket for a Jetpack site, if the jetpack site has add-ons (and no other plan) the ticket plan in Zendesk will be `add-on`, and the add-ons will appear in the Jetpack Add-ons field (see screenshots below). Note that if the Jetpack site has a plan, this will prevail over the add-ons.
3. If the current site is a self hosted site, a `selected_site_self_hosted` tag will be added to the request



Zendesk Plan | Zendesk Jetpack Add-ons
-- | --
<img width="400" src="https://user-images.githubusercontent.com/34376330/123871322-3609cb80-d8f9-11eb-95c7-38a10ba734e4.png"> | <img width="400" src="https://user-images.githubusercontent.com/34376330/123871330-399d5280-d8f9-11eb-91f3-f47b5da5b566.png">

Zendesk ticket list
<img width="900" src="https://user-images.githubusercontent.com/34376330/124656782-3f4de780-de67-11eb-872b-0580b6352538.png">


**Note: please don't forget to delete any ticket that was created to test this PR once you are finished reviewing. Thank you.**
To test:
Pre requisites:
- Have an account with at least a self-hosted, jetpack connected site with one or more add-ons (e.g. Jetpack Scan realtime) and one or more wp.com sites.
- Build/run and select the jetpack site on my site
- Open a support request, then go to Zendesk and make sure that the Add-ons show up in the ticket as in the above screenshots.
- Switch site (either wp.com or another Jetpack site with a paid plan, e.g. premium), open another request and make sure that the plan in the ticket always matches the selected site
- Make sure that any request you open from a self hosted site (jetpack connected or not) contains the tag `mobile_self_hosted`

## Regression Notes
1. Potential unintended areas of impact
Limited, this change only affects the plan and add-on fields sent to Zendesk

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Test the feature with a variety of combinations of sites/plans/add-ons

3. What automated tests I added (or what prevented me from doing so)
none
PR submission checklist:
